### ISSUE TITLE
`:used-underscore-binding` reports false positives with `clojure.core.match/match`

### ISSUE THREAD

**Issue by camsaul** [clj-kondo/clj-kondo#2152]
**version**

```
clj-kondo v2023.07.13
```

**macro usage**

Yes, `clojure.core.match/match`

**platform**

Native version installed via Homebrew on Linux

**editor**

N/A

**problem**

If you enable the `:used-underscored-binding` linter, a form like

```clj
(clojure.core.match/match [:name x y]
  [:name _ _]
  "Ok")
```

triggers a 

```
Used binding is marked as unused: _
```

warning. This is a a false positive, since `_` has a special meaning in `clojure.core.match/match`. It's not used in the macroexpansion either:

```clj
(let [ocr-149912 :name x x y y]
  (try
    (cond (= ocr-149912 :name) "Ok" :else (throw clojure.core.match/backtrack))
    (catch Exception e__46989__auto__
      (if (identical? e__46989__auto__ clojure.core.match/backtrack)
        (throw (java.lang.IllegalArgumentException. (str "No matching clause: " ocr-149912 " " x " " y)))
        (throw e__46989__auto__)))))
```

**repro**

```clj
(ns repro
  (:require [clojure.core.match :as match]))

(let [x 1 y 2]
  (match/match [:name x y]
    [:name _ _]
    "Ok"))
```

**config**

```clj
{:linters {:used-underscored-binding {:level :warning}}}
```

**expected behavior**

Since this is a part of Clojure core, it would be nice if it worked out of the box. If that's not possible, it would be nice if the `:used-underscore-binding` linter was configurable and could be disabled inside of specific forms `clojure.core.match` forms without needing a `:clj-kondo/ignore` every time you use it

**Comment by borkdude** [comment]
Thanks, makes sense
